skyportal/tests/frontend/test_top_sources.py

```diff
@@ -39,3 +41,50 @@ def test_top_sources(driver, user, public_source, public_group, upload_data_toke
     assert status == 200
     driver.refresh()
     driver.wait_for_xpath("//*[contains(.,'2 view(s)')]")
+
+
+@pytest.mark.flaky(reruns=2)
+def test_top_source_prefs(driver, user, public_source, public_group, upload_data_token):
+    # Add an old source and give it an old view
+    obj_id = str(uuid.uuid4())
+    status, data = api(
+        'POST',
+        'sources',
+        data={
+            'id': obj_id,
+            'ra': 50.4,
+            'dec': 22.33,
+            'redshift': 2.1,
+            "altdata": {"simbad": {"class": "RRLyr"}},
+            'transient': False,
+            'ra_dis': 2.3,
+            'group_ids': [public_group.id],
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert data['data']['id'] == obj_id
+
+    twenty_days_ago = datetime.datetime.now() - datetime.timedelta(days=20)
+    sv = SourceView(
+        obj_id=obj_id,
+        username_or_token_id=upload_data_token,
+        is_token=True,
+        created_at=twenty_days_ago,
+    )
+    DBSession.add(sv)
```

Comment:
Why call the session factory directly instead of `DBSession()`?

Reply:
Oops, yeah oversight on my part.